internal/wrappers/features/step_defs_cpp/realtime-steps.cpp

```diff
@@ -0,0 +1,114 @@
+#include "steps.hpp"
+
+// Anonymous namespace to handle a linker error
+// see https://stackoverflow.com/questions/14320148/linker-error-on-cucumber-cpp-when-dealing-with-multiple-feature-files
+namespace {
+
+  GIVEN("^I subscribe to \'(.*)\'$") {
+    REGEX_PARAM(std::string, collection_id);
+   
+    ScenarioScope<KuzzleCtx> ctx;
+
+    ctx->listener = new CustomNotificationListener();
+
+    try {
+      ctx->room_id = ctx->kuzzle->realtime->subscribe(ctx->index, collection_id, "{}", ctx->listener);
+    } catch (KuzzleException e) {
+      BOOST_FAIL(e.getMessage());
+    }
+  }
+
+  WHEN("^I create a document in ([^\"]*)$") {
+    REGEX_PARAM(std::string, collection_id);
+
+    ScenarioScope<KuzzleCtx> ctx;
+
+    query_options options = {0};
+    options.refresh = (char *)"wait_for";
+
+    try {
+      ctx->kuzzle->document->create(ctx->index, ctx->collection, "", "{\"foo\":\"bar\"}", &options);
+    } catch (KuzzleException e) {
+      BOOST_FAIL(e.getMessage());
+    }
+  }
+
+  THEN("^I receive a notification$") {
+    ScenarioScope<KuzzleCtx> ctx;
+    sleep(1);
+    BOOST_CHECK(ctx->notif_result != NULL);
+    ctx->notif_result = NULL;
+    ctx->kuzzle->realtime->unsubscribe(ctx->room_id);
+    delete ctx->listener;
+    delete ctx->notif_result;
```

Comment:
I would set those to ref to null, so that no one attempt to use them in other steps.